inttests/volume_test.go

```diff
@@ -372,6 +372,78 @@ func TestMapQueryUnmapVolume(t *testing.T) {
 	deleteAllVolumes(t)
 }
 
+func TestMapQueryUnmapSnapshot(t *testing.T) {
+	volID, err := createVolume(t, "")
+	assert.Nil(t, err)
+	newVolume, err := getVolByID(volID)
+	assert.Nil(t, err)
+
+	// now make a snapshot
+	snapshotDefs := make([]*siotypes.SnapshotDef, 0)
+	snapname := fmt.Sprintf("%s-%s", newVolume.Name, "snap")
+
+	snapDef := &siotypes.SnapshotDef{
+		VolumeID:     volID,
+		SnapshotName: snapname,
+	}
+	snapshotDefs = append(snapshotDefs, snapDef)
+	snapParam := &siotypes.SnapshotVolumesParam{
+		SnapshotDefs: snapshotDefs,
+	}
+
+	system := getSystem()
+	assert.NotNil(t, system)
+
+	// Create snapshot
+	snapResponse, err := system.CreateSnapshotConsistencyGroup(snapParam)
+	assert.Nil(t, err)
+	assert.NotZero(t, len(snapResponse.VolumeIDList))
+
+	// Get StoragePool
+	pool := getStoragePool(t)
+	volumes, err := pool.GetVolume("", volID, "", "", true)
+	assert.Nil(t, err)
+	assert.NotNil(t, volumes)
+
+	// Get Snapshot
+	volumes, err = pool.GetVolume("", snapResponse.VolumeIDList[0], "", "", true)
+	assert.Nil(t, err)
+	assert.NotNil(t, volumes)
+
+	// Lock Snapshot
+	snap, err := getVolByID(volumes[0].ID)
+	assert.Nil(t, err)
+	sr := goscaleio.NewVolume(C)
+	sr.Volume = snap
+	sr.SetVolumeAccessModeLimit("ReadWrite")
```

Comment:
Can you please cover some negative test scenarios as well.

Reply:
![image](https://user-images.githubusercontent.com/118425422/207309123-5f8f40be-84b3-430d-bc06-4b9cf5f39d0a.png)
added the negative test for the above scenario.